azure-quantum/azure/quantum/qiskit/backends/backend.py

```diff
@@ -75,6 +75,17 @@ def _translate_input(self, circuit, data_format, input_params):
         if not "arguments" in input_params:
             input_params["arguments"] = []
 
+        # By default, we'll transpile into the list of supported instructions unless the user
+        # has indicated that this step should be skipped.
+        auto_transpile = not input_params["skip-auto-transpile"] if "skip-auto-transpile" in input_params else True
+
+        if auto_transpile:
+            from qiskit import transpile
+            circuit = transpile(circuit, self)
+
+            if logger.isEnabledFor(logging.DEBUG):
```

Comment:
Is this condition used to prevent overhead of calling `to_qir`? If so, a comment to that effect may be helpful.